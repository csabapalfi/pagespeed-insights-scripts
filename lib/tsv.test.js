const {results, stats} = require('./tsv');

describe('tsv formatter', () => {
  describe('results', () => {
    const result = {
      type: 'result',
      fetchTime: '21:48:05',
      score: 95,
      FCP: 0.9,
      FMP: 1,
      SI: 1.1,
      FCI: 3.2,
      TTI: 4
    };

    it('are formatted as expected', () => {
      expect(results([result]))
        .toMatchSnapshot();
    });
  
    it('heading row added before first result', () => {
      expect(results([result], true))
        .toMatchSnapshot();
    });

    it('userTimingMark uses default number format precision (2 digits)', () => {
      expect(results([{...result, DPA: 1.25}]))
        .toMatchSnapshot();
    });
  });

  describe('stats', () => {
    const statistic = {
      type: 'statistic',
      name: 'median',
      score: 95,
      FCP: 0.9,
      FMP: 1,
      SI: 1.1,
      FCI: 3.2,
      TTI: 4
    };
   
    it('stats are formatted as expected', () => {
      expect(stats([statistic]))
        .toMatchSnapshot();
    });

    it('stddev uses custom precision for score (1 digit)', () => {
      expect(stats([{...statistic, name: 'stddev'}]))
        .toMatchSnapshot();
    });
  });

});