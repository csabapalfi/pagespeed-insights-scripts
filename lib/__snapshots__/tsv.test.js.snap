// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tsv formatter results are formatted as expected 1`] = `
Array [
  "21:48:05	95	0.9	1.0	1.1	3.2	4.0",
]
`;

exports[`tsv formatter results heading row added before first result 1`] = `
Array [
  "fetchTime	score	FCP	FMP	SI	FCI	TTI",
  "21:48:05	95	0.9	1.0	1.1	3.2	4.0",
]
`;

exports[`tsv formatter results userTimingMark uses default number format precision (2 digits) 1`] = `
Array [
  "21:48:05	95	0.9	1.0	1.1	3.2	4.0	1.25",
]
`;

exports[`tsv formatter stats stats are formatted as expected 1`] = `
Array [
  "",
  "median  	95	0.9	1.0	1.1	3.2	4.0",
]
`;

exports[`tsv formatter stats stddev uses custom precision for score (1 digit) 1`] = `
Array [
  "",
  "stddev  	95.0	0.9	1.0	1.1	3.2	4.0",
]
`;
